codalab/worker/state_committer.py

```diff
@@ -26,7 +28,14 @@ def load(self, default=None):
 
     def commit(self, state):
         """ Write out the state in JSON format to a temporary file and rename it into place """
-        with tempfile.NamedTemporaryFile() as f:
-            f.write(pyjson.dumps(state).encode())
-            f.flush()
-            shutil.copyfile(f.name, self._state_file)
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            try:
+                f.write(pyjson.dumps(state).encode())
+                f.flush()
+                shutil.copyfile(f.name, self._state_file)
+            finally:
+                try:
+                    os.unlink(f.name)
+                except FileNotFoundError:
+                    logging.info("Problem occurred in creating temp file {}".format(f.name))
```

Comment:
This error is catching when calling `os.unlink`. I think the log message should mention that instead.